### ISSUE TITLE
Broken public API for mouse click event without deprecation

### ISSUE THREAD

**Issue by Czaki** [napari/napari#7054]
### 🐛 Bug Report

After #5432 there is no public API to mouse click callback. 
Before this PR the `viewer_widget.canvas.events.mouse_press` could be used. Now it will be `viewer_widget.canvas._scene_canvas.events.mouse_press`

The `MousemapProvider` also does not contain a place for such a callback. 

https://github.com/napari/napari/blob/f0c02a844cdd0eebceb64b696bb4ab73c2aed08f/napari/utils/mouse_bindings.py#L23-L30

Also, the current implementation of MousemapProvider is not safe and may keep object alive to long (hard references).

### 💡 Steps to Reproduce

.

### 💡 Expected Behavior

I think that we should provide `event` property to `VispyCanvas` that will provide a backward compatibility layer. Then we should decide if it should be deprecated or no.

If we decide that `MousemapProvider` is the best place for a stable API then we should reimplement it to store weak references to methods. (maybe using psygnal)

### 🌎 Environment

0.5.0a4

**Comment by jni** [comment]
@Czaki I'm confused about

> The MousemapProvider also does not contain a place for such a callback.

The callbacks list are such a place? I just tried the zarpaint plugin and it works:

https://github.com/jni/zarpaint/blob/5d4c8508e731ca8417e0ecbd44c5bf8079ff7a3a/src/zarpaint/_add_3d_points.py#L86-L104

using `Layer.mouse_drag_callbacks.append`.

**Comment by Czaki** [comment]
> The callbacks list are such a place? I just tried the zarpaint plugin and it works:

Yes. On list you store hard references. In both, our events and psygnal implementation there is special code to use weak references, so events will not prevent from garbage collecting object.

**Comment by jni** [comment]
Sounds like a long-standing issue that can be solved after 0.5.0?

**Comment by Czaki** [comment]
> Sounds like a long-standing issue that can be solved after 0.5.0?

This part, yes. Lack of mouse click callback, no.

**Comment by jni** [comment]
GH code search suggests only PartSeg uses this API. 😅 (All other hits were old napari forks or vispy users.) Do you think you could use the private API while we come up with a fix? 😅 😬

**Comment by Czaki** [comment]
> It seems though that the API is broken altogether without a deprecation warning?

yes



> * add a property that restores the old path
> 
> * release
> 
> * discuss what the new API should be
> 
> * deprecate property in favour of new path
> 
> * remove deprecated property

good plan

**Comment by jni** [comment]
#7060 implemented the first part of [this plan](https://github.com/napari/napari/issues/7054#issuecomment-2205548968). I think it's not urgent to finish the plan so I've marked this as 0.5 milestone, but we can revisit any time this year really.

**Comment by psobolewskiPhD** [comment]
I was helping Lars at SciPy with a demo and we were super confused about the lack of `mouse_click_callbacks`.
Turns out you can hack it using the drag one, but it's not obvious.

**Comment by jni** [comment]
Yeah I remember when we unified things feeling a bit icky, so I agree with you. 😂 But yeah a click is just a drag with no movement. 😂 I wouldn't be opposed to adding back a shortcut...

**Comment by psobolewskiPhD** [comment]
well the issue is there is no docs, so it's annoying to figure out that a click is a drag...
